lib/src/simulations/rose_pattern.dart

```diff
@@ -154,32 +198,139 @@ class _RosePatternState extends State<RosePattern> {
                 'k ~ ${(_n / _d).toStringAsFixed(2)}',
                 style: Theme.of(context).textTheme.subtitle,
               ),
-            )
+            ),
+            Positioned(
+              top: 0,
+              right: 0,
+              child: Row(
+                mainAxisAlignment: MainAxisAlignment.end,
+                children: <Widget>[
+                  Text("Animate"),
+                  Checkbox(
+                    onChanged: (_){
+                      setState(() {
+                        animate=!animate;
+                        if(animating)
+                          animating = (animating && animate);
+                      });
+                    },
+                    value: animate,
+                    activeColor: Colors.red,
+                  )
+                ],
+              ),
+            ),
           ],
         ),
       ),
     );
   }
 }
 
+class Rose extends StatefulWidget {
+  Rose({
+    Key key,
+    @required this.d,
+    @required this.n,
+    @required this.c,
+    @required this.animate,
+    @required this.animating
+  }) :super(key:key);
+
+  final double d;
+  final double n;
+  final double c;
+  final bool animate;
+  final bool animating;
+  @override
+  _RoseState createState() => _RoseState();
+}
+
+class _RoseState extends State<Rose> {
+  List<Offset> points = [];
+  double loopi = 0;
+  double r, k;
+  double looplength = 2 * pi;
+
+
+  void dispose() {
+    super.dispose();
+  }
+
+  void clearScreen() {
+    points.clear();
+    looplength = 2 * pi * widget.d;
+    loopi=0;
+  }
+
+  nextStep()
+  {
+    if(loopi>=looplength)
+    {
+      clearScreen();
+      looplength = 2 * pi * widget.d;
+      loopi = 0;
+    }
+
+    setState(() {
+      if(!(widget.d==0)&&!(widget.d==0&&widget.n==0)){
+        sleep(Duration(milliseconds: 1));
```

Comment:
Can you make the animation a bit faster while keeping it smooth? I tried changing these values but couldn't find a good match.